### PR TITLE
fix: add snapshot related commands in must-gather

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Add commands in must-gather to capture snapshot related
kubernetes objects.

Signed-off-by: Mudit Agarwal <muagarwa@redhat.com>

### CODE REVIEW

**Comment by agarwal-mudit** [comment]
/hold
I still need Yug's changes to test this.

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (nbalacha) [inline must-gather/collection-scripts/gather_namespaced_resources:55]: Are the volumeSnapshot resources created in the same namespace as the PVC? If yes, this will not be the openshift-storage namespace.

**Comment by agarwal-mudit** [comment]
Tested on a cluster without snapshots, it is not giving any error.
snapshots are not present so it is just saying **`No resources found`**

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (nbalacha) [inline must-gather/collection-scripts/gather_namespaced_resources:56]: I believe the volumesnapshotclass is not a namespaced resource.  It should be a clusterscoped resource.

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (nbalacha) [inline must-gather/collection-scripts/gather_namespaced_resources:67]: Will this be created in the install namespace or the PVC's namespace?

**Review by nbalacha — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (agarwal-mudit) [reply must-gather/collection-scripts/gather_namespaced_resources]: Fixed

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (agarwal-mudit) [reply must-gather/collection-scripts/gather_namespaced_resources]: Fixed

File: must-gather/collection-scripts/gather_namespaced_resources
Comment (agarwal-mudit) [reply must-gather/collection-scripts/gather_namespaced_resources]: fixed

**Review by nbalacha — COMMENTED** [review]
The code changes look fine and I will approve it once it has been tested against the snapshot PR.

**Comment by agarwal-mudit** [comment]
[
[must-gather.local.1353291690104036139.zip](https://github.com/red-hat-storage/lvm-operator/files/8472777/must-gather.local.1353291690104036139.zip)
](url)

**Comment by agarwal-mudit** [comment]
@nbalacha I have uploaded the must-gather for the changed code, please take a look.

**Comment by nbalacha** [comment]
> @nbalacha I have uploaded the must-gather for the changed code, please take a look.

Looks good.

**Review by nbalacha — APPROVED** [review]

**Pull request merged**
